packages/dashboard/src/utils/dash.ts

```diff
@@ -59,14 +70,74 @@ export async function confirmMessage(
     await lifecycle.respond({ payload });
   } catch (err: any) {
     const muteErrPattern =
-      /A response has already been sent for message id.*of type "provider"./;
+      /^A response has already been sent for message id .* of type "provider"\.$/;
     if (!muteErrPattern.test(err.message)) {
       console.error(err);
     }
   }
   return payload;
 }
 
+export async function decodeMessage(
+  lifecycle: ReceivedMessageLifecycle<DashboardProviderMessage>,
+  decoder: ProjectDecoder
+) {
+  const method = lifecycle.message.payload.method as DecodableRpcMethod;
+
+  switch (method) {
+    case "eth_sendTransaction": {
```

Comment:
Something to consider for future work - do we want to decode/display pre-signed messages (`eth_sendRawTransaction`)? Or maybe we even want to prevent them from getting sent through and tell the user to use the browser wallet for better key management?

Reply:
From a cursory search I'm unable to verify that MM is open to broadcasting an `eth_sendRawTransaction` request, but I will note this down, thanks.

Maybe we can do both! Display it and show users a notification about wallet safekeeping.